instrumentation/jms/src/main/java/brave/jms/PropertyFilter.java

```diff
@@ -74,6 +77,31 @@ static void filterProperties(Message message, Set<String> namesToClear, List<Obj
       return;
     }
 
+    // workaround for ActiveMQBytesMessage
+    if (message instanceof BytesMessage) { // BytesMessage requires clearing the body before reset properties otherwise it fails with MessageNotWriteableException
+      resetBytesMessageProperties(message, out);
+    } else {
+      resetProperties(message, out);
+    }
+  }
+
+  private static void resetBytesMessageProperties( Message message, List<Object> out ) {
```

Comment:
this will cause a lot of overhead as we are subject to OOM. as this is a bug in activemq, we need to be more specific than this.